### ISSUE TITLE
pikaur crashes when updating libc++ and libc++abi

### ISSUE THREAD

**Issue by lectrode** [actionless/pikaur#233]
Output:
```
$ sudo pikaur -Su
:: Starting full AUR upgrade...
Reading repository package databases...
Reading local package database...
Reading AUR packages info...
Resolving AUR dependencies...
Traceback (most recent call last):
  File "/usr/bin/pikaur", line 8, in <module>
    main()
  File "/usr/lib/python3.6/site-packages/pikaur/main.py", line 309, in main
    cli_entry_point()
  File "/usr/lib/python3.6/site-packages/pikaur/main.py", line 229, in cli_entry_point
    cli_install_packages(args)
  File "/usr/lib/python3.6/site-packages/pikaur/main.py", line 125, in cli_install_packages
    InstallPackagesCLI(args=args)
  File "/usr/lib/python3.6/site-packages/pikaur/install_cli.py", line 147, in __init__
    self.get_all_packages_info()
  File "/usr/lib/python3.6/site-packages/pikaur/install_cli.py", line 199, in get_all_packages_info
    manually_excluded_packages_names=self.manually_excluded_packages_names,
  File "/usr/lib/python3.6/site-packages/pikaur/install_info_fetcher.py", line 44, in __init__
    self.get_all_packages_info()
  File "/usr/lib/python3.6/site-packages/pikaur/install_info_fetcher.py", line 102, in get_all_packages_info
    self.get_aur_deps_info()
  File "/usr/lib/python3.6/site-packages/pikaur/install_info_fetcher.py", line 292, in get_aur_deps_info
    self.aur_deps_relations = find_aur_deps(all_aur_packages_names)
  File "/usr/lib/python3.6/site-packages/pikaur/aur_deps.py", line 234, in find_aur_deps
    results = request.get()
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
  File "/usr/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/usr/lib/python3.6/site-packages/pikaur/aur_deps.py", line 132, in find_missing_deps_for_aur_pkg
    source=PackageSource.LOCAL
  File "/usr/lib/python3.6/site-packages/pikaur/aur_deps.py", line 101, in check_deps_versions
    del deps[dep_name]
KeyError: 'libc++abi'
```

**Comment by lectrode** [comment]
The bottom of the output mentioned "KeyError", which may be gpg related. But even if pikaur does not currently have the ability to add gpg keys, it probably shouldn't crash when it encounters a package that needs it. But that's just a guess.

**Comment by gourdcaptain** [comment]
I can verify this happening for me as well, and libc++'s associated gpg keys were already added to my keyring (and work with other means of building the package.)

**Comment by lectrode** [comment]
Possibly related: `pikaur` also fails to install them even if I have the gpg keys imported:
```
$ gpg --recv-keys 474E22316ABF4785A88C6E8EA2C794A986419D8A
gpg: key A2C794A986419D8A: "Tom Stellard <tstellar@redhat.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1

$ gpg --recv-keys B6C8F98282B944E3B0D5C2530FC3042E345AD05D
gpg: key 0FC3042E345AD05D: 3 signatures not checked due to missing keys
gpg: key 0FC3042E345AD05D: "Hans Wennborg <hans@chromium.org>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1

$ sudo pikaur -S --noconfirm libc++ libc++abi
[sudo] password for lectrode: 
Reading repository package databases...
Reading local package database...
Resolving AUR dependencies...
looking for conflicting AUR packages...
:: Skipping review of PKGBUILD for libc++, libc++abi package (--noconfirm)


:: Starting the build:
Running as unit: run-u4790.service
==> Making package: libc++ 6.0.1-1 (Wed 11 Jul 2018 12:50:00 PM PDT)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Downloading llvm-6.0.1.src.tar.xz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 24.1M  100 24.1M    0     0   500k      0  0:00:49  0:00:49 --:--:--  624k
  -> Downloading llvm-6.0.1.src.tar.xz.sig...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   543  100   543    0     0   1479      0 --:--:-- --:--:-- --:--:--  1479
  -> Downloading libcxx-6.0.1.src.tar.xz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 1515k  100 1515k    0     0   859k      0  0:00:01  0:00:01 --:--:--  859k
  -> Downloading libcxx-6.0.1.src.tar.xz.sig...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   543  100   543    0     0   1798      0 --:--:-- --:--:-- --:--:--  1798
  -> Downloading libcxxabi-6.0.1.src.tar.xz...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  515k  100  515k    0     0   824k      0 --:--:-- --:--:-- --:--:--  822k
  -> Downloading libcxxabi-6.0.1.src.tar.xz.sig...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   543  100   543    0     0   1898      0 --:--:-- --:--:-- --:--:--  1898
==> Validating source files with sha512sums...
    llvm-6.0.1.src.tar.xz ... Passed
    llvm-6.0.1.src.tar.xz.sig ... Skipped
    libcxx-6.0.1.src.tar.xz ... Passed
    libcxx-6.0.1.src.tar.xz.sig ... Skipped
    libcxxabi-6.0.1.src.tar.xz ... Passed
    libcxxabi-6.0.1.src.tar.xz.sig ... Skipped
==> Verifying source file signatures with gpg...
    llvm-6.0.1.src.tar.xz ... FAILED (unknown public key A2C794A986419D8A)
    libcxx-6.0.1.src.tar.xz ... FAILED (unknown public key A2C794A986419D8A)
    libcxxabi-6.0.1.src.tar.xz ... FAILED (unknown public key A2C794A986419D8A)
==> ERROR: One or more PGP signatures could not be verified!
Finished with result: exit-code
Main processes terminated with: code=exited/status=1
Service runtime: 54.506s
Command 'systemd-run --pipe --wait -p DynamicUser=yes -p CacheDirectory=pikaur -E HOME=/tmp -p WorkingDirectory=/var/cache/pikaur/build/libc++ makepkg --force' failed to execute.
Do you want to retry? [N]o (--noconfirm)


Can't build 'libc++'.

Failed to build following packages:
libc++
libc++abi

```

**Comment by gourdcaptain** [comment]
Did you add them to the user gpg set or the system pacman set? Because it should be user one for makepkg, and the output where it fails you posted makepkg output that shouldn't be affected by pikaur.

**Comment by lectrode** [comment]
Running `pikaur -S --noconfirm libc++ libc++abi` (without sudo) works. The last one only fails with sudo. Original crash still happens with or without sudo. I'll create a separate issue for that failed install with sudo if you think it's a separate issue. I kinda think it is.

**Comment by gourdcaptain** [comment]
The reason the last one fails with sudo (I think) is you're trying to build the package as root which means it's not running as the user who's added the keys to their keyring. Also running pikaur as root directly is a bad idea, IMHO, it'll escalate itself and ask for the password when needed.

**Comment by lectrode** [comment]
Thank you for the warning. I'm aware of the implications of running it with sudo. I've removed the output from the other issue, since it is not related to the original issue.

**Comment by Morganamilo** [comment]
I would assume key error refers to a key in a python dictionary.

**Comment by actionless** [comment]
@lectrode  please update to the latest `pikaur-git` and try again

**Comment by actionless** [comment]
tagged 0.15 since there are few more bugfixes pending on master

thanks for checking!
